app/controllers/api/automate_domains_controller.rb

```diff
@@ -1,5 +1,29 @@
 module Api
   class AutomateDomainsController < BaseController
+    REQUIRED_FIELDS = %w[git_url ref_type ref_name].freeze
+
+    def create_from_git_resource(type, _id, data)
+      assert_all_required_fields_exists(data, type, REQUIRED_FIELDS)
+      raise BadRequestError, 'ref_type must be "branch" or "tag"' unless valid_ref_type?(data)
+
+      api_log_info("Create will be queued for automate domain from #{data["git_url"]} / #{data["ref_name"]}")
+
+      begin
+        unless GitBasedDomainImportService.available?
+          raise "Git owner role must enabled to import git repositories"
+        end
+        task_id = GitBasedDomainImportService.new.queue_refresh_and_import(data["git_url"],
+                                                                           data["ref_name"],
+                                                                           data["ref_type"],
+                                                                           User.current_user.current_tenant.id,
+                                                                           prepare_optional_auth(data))
+
+        action_result(true, 'Creating Automate Domain from git repository', :task_id => task_id)
```

Comment:
can we add some identifying text here ?  Creating Automate Domain XYZZY from git repository,
maybe using ref_name.

Reply:
@abellotti Great idea! Thank you. I think the `ref_name` alone might not be very helpful since it could simply be the branch name  `master`.   I think both the `git_url\ref_name`.
Please let me know if you are OK with this.